warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -581,12 +584,26 @@ public void seek(Range r, Collection<ByteSequence> columnFamilies, boolean inclu
                     querySpan = ((SourceTrackingIterator) source).getQuerySpan();
                 }
 
-                // seek our underlying source to the start of the incoming range
-                // expand the range as the underlying table may not actually contain the keys in this range as we are only returning keys
-                // as specified by the returnKeyType
-                Range seekRange = new Range(lastRangeSeeked.getStartKey(), lastRangeSeeked.isStartKeyInclusive(),
-                                (lastRangeSeeked.getEndKey() == null ? null : new Key(lastRangeSeeked.getEndKey().getRow()).followingKey(PartialKey.ROW)),
-                                false);
+                // seek the source to the start of the field index that matches the specified field
+                // this initial seek ensures that data for the field is actually present
+                // if data is present, then bounding ranges are generated
+                Preconditions.checkNotNull(fiRow, "Ivarator row was null");
+                Preconditions.checkNotNull(fieldName, "Ivarator was not bounded by a field");
+
+                Range seekRange;
+                if (fieldName.toString().equals(Constants.ANY_FIELD)) {
+                    // in the case of an ANYFIELD, which should only happen during unit tests, restrict the scan range
+                    // to just the fi prefix.
+                    Key start = new Key(this.fiRow, new Text("fi\0"));
+                    Key stop = new Key(this.fiRow, new Text("fi\0\uffff"));
+                    seekRange = new Range(start, lastRangeSeeked.isStartKeyInclusive(), stop, false);
```

Comment:
Same line (seekRange = ...) in both branches -- could go after if/else

Reply:
Done